File: MoodByte/src/main/java/org/example/ModeloDTO/EjercicioDTO.java
package org.example.ModeloDTO;

import org.example.Modelo.Ejercicio;

import java.sql.Time;

public record EjercicioDTO (
        long id,
        String titulo,
        String descripcion,
        String recursoUrl,
        Time duracion,
        long idEstado
){
    public EjercicioDTO(Ejercicio ejercicio){
        this(
          ejercicio.getId(),
          ejercicio.getTitulo(),
          ejercicio.getDescripcion(),
          ejercicio.getRecursoUrl(),
          ejercicio.getDuracion(),
          ejercicio.getEstado().getId()
        );
    }
}

File: MoodByte_Interfaz/ModeloDTO/EjercicioDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ModeloDTO
{
    public class EjercicioDTO
    {
        public long id { get; set; }
        public string titulo { get; set; }
        public string descripcion { get; set; }
        public string recursoUrl { get; set; }
        public TimeOnly duracion { get; set; }
        public long idEstado { get; set; }
    }
}
